doc/source/tokudb/tokudb_variables.rst

```diff
@@ -4,473 +4,1465 @@
+  :vartype: Numeric
 
-.. variable:: tokudb_disable_slow_alter
+This variable configures the size in bytes of the |TokuDB| cache table. The
+default cache table size is 1/2 of physical memory. Percona highly recommends
+using the default setting if using buffered I/O, if using direct I/O then
+consider setting this parameter to 80% of available memory.
 
-This variable controls whether slow alter tables are allowed. For example, the following command is slow because HCADER does not allow a mixture of column additions, deletions, or expansions:
+Consider decreasing :variable:`tokudb_cache_size` if excessive swapping is
+causing performance problems. Swapping may occur when running multiple |MySQL|
+server instances or if other running applications use large amounts of physical
+memory.
 
-.. code-block:: mysql
+.. variable:: tokudb_cachetable_pool_threads
 
-  ALTER TABLE table
-  ADD COLUMN column_a INT, 
-  DROP COLUMN column_b;
+  :cli: Yes
+  :conf: Yes
+  :scope: Global
+  :dyn: No
+  :vartype: Numeric
+  :range: 0 - 1024
+  :default: 0
+
+This variable defines the number of threads for the cachetable worker thread
+pool. This pool is used to perform node prefetches, and to serialize, compress,
+and write nodes during cachetable eviction. The default value of 0 calculates
+the pool size to be num_cpu_threads * 2.
+
+.. variable:: tokudb_check_jemalloc
+
+  :cli: Yes
+  :conf: Yes
+  :scope: Global
+  :dyn: No
+  :vartype: Boolean
+  :default: ON
+
+This variable enables/disables startup checking if jemalloc is linked and
+correct version and that transparent huge pages are disabled. Used for
+testing only.
+
+.. variable:: tokudb_checkpoint_lock
+
+  :cli: Yes
+  :conf: Yes
+  :scope: Session, Global
+  :dyn: Yes
+  :vartype: Boolean
+  :default: OFF
+
+Disables checkpointing when true. Session variable but acts like a global, any
+session disabling checkpointing disables it globally. If a session sets this
+lock and disconnects or terminates for any reason, the lock will not be
+released. Special purpose only, do **not** use this in your application.
+
+.. variable:: tokudb_checkpoint_on_flush_logs
+
+  :cli: Yes
+  :conf: Yes
+  :scope: Global
+  :dyn: Yes
+  :vartype: Boolean
+  :default: OFF
+
+When enabled forces a checkpoint if we get a flush logs command from the
+server.
 
-By default, :variable:`tokudb_disable_slow_alter` is disabled, and the engine reports back to mysql that this is unsupported resulting in the following output:
+.. variable:: tokudb_checkpoint_pool_threads
 
-.. code-block:: bash
+  :cli: Yes
+  :conf: Yes
+  :scope: Global
+  :dyn: No
+  :vartype: Numeric
+  :range: 0 - 1024
+  :default: 0
 
-   ERROR 1112 (42000): Table 'test_slow' uses an extension that doesn't exist in this MySQL version
+This defines the number of threads for the checkpoint worker thread pool. This
+pool is used to serialize, compress and write nodes cloned during checkpoint.
+Uses old algorithm to set pool size to 4 / num_cpu_threads.
```

Comment:
Hrvoje pointed out this is incorrect. The correct default formula is num_cpu_threads / 4, not 4 / num_cpu_threads or num_cpu_threads * 2.